src/server/pps/server/api_server.go

```diff
@@ -189,7 +189,7 @@ func (a *apiServer) CreateJob(ctx context.Context, request *pps.CreateJobRequest
 	defer func(start time.Time) { metricsFn(start, retErr) }(time.Now())
 
 	job := &pps.Job{uuid.NewWithoutUnderscores()}
-	sort.Slice(request.Inputs, func(i, j int) bool { return request.Inputs[i].Name < request.Inputs[j].Name })
```

Comment:
What does using `SliceStable` here get us? It seems like maintaining the order of the inputs shouldn't matter if all of the names are unique which is a requirement right?